dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/impl/WSRdbManagedConnectionImpl.java

```diff
@@ -3866,15 +3858,17 @@ public final void setAutoCommit(boolean value) throws SQLException
 
     public final void setTransactionIsolation(int isoLevel) throws SQLException 
     {
-        if (currentTransactionIsolation != isoLevel) {
-            // TODO Determine behavior when switching from TRANSACTION_NONE to another isolation level
-//            if ( isDSTranNone && (isoLevel != Connection.TRANSACTION_NONE) ) {
-//                throw new SQLException(AdapterUtil.getNLSMessage("DSRA4011.tran.none.iso.switch.unsupported", dsConfig.get().id));
-//            } 
-            
-            // Reject switching to Transaction_None
+        if (currentTransactionIsolation != isoLevel) {            
+            // Reject switching to an isolation level of TRANSACTION_NONE unless 
+            // the driver is configured with an isolation level of TRANSACTION_NONE.
             if (isoLevel == Connection.TRANSACTION_NONE) {
-                throw new SQLException(AdapterUtil.getNLSMessage("DSRA4011.tran.none.iso.switch.unsupported", dsConfig.get().id));
+                if(sqlConn.getTransactionIsolation() != Connection.TRANSACTION_NONE) {
+                    throw new SQLException(AdapterUtil.getNLSMessage("DSRA4011.tran.none.iso.switch.unsupported", dsConfig.get().id));
+                } else {
+                    currentTransactionIsolation = isoLevel;
+                    isolationChanged = true;
```

Comment:
It seems kind of wrong to set isolationChanged to true, since we haven't really changed the iso level?  Is there a need to set this variable?